src/sentry/sentry_metrics/consumers/indexer/routing_producer.py

```diff
@@ -0,0 +1,166 @@
+import logging
+import time
+from abc import ABC, abstractmethod
+from collections import deque
+from concurrent.futures import Future
+from typing import Any, Deque, MutableMapping, NamedTuple, Optional, Tuple
+
+from arroyo.backends.kafka import KafkaPayload
+from arroyo.processing.strategies import ProcessingStrategy
+from arroyo.types import Commit, Message, Partition, Position, Topic
+from confluent_kafka import Producer
+
+logger = logging.getLogger(__name__)
+
+
+class RoutingPayload(NamedTuple):
+    """
+    Payload suitable for the ``RoutingProducer``. ``MessageRouter`` works
+    with this payload type. The routing_headers are used to determine the
+    route for the message. The payload is the message body which should be sent
+    to the destination topic.
+    """
+
+    routing_header: MutableMapping[str, Any]
+    routing_message: KafkaPayload
```

Comment:
Who fills in the routing headers? I'm guessing this is the same data that is in `KafkaPayload.headers`? Do we really want to make another copy of all the headers here? How about just replacing `routing_header` with `org_id` (or `routing_key` as a more generic option) or something.

Reply:
The processing step before this step will fill in the routing headers. The routing header is different from the message headers. The message headers are not present in the routing header. You can see an example of how routing header would be used over here https://github.com/getsentry/sentry/pull/40776/files#diff-c7165353f1e7e8138f5624281fb19b84e9c5ac50d8eead26974bb0666ed1f4d1R27-R34